src/cpp/src/image_generation/models/flux_transformer_2d_model.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/image_generation/flux_transformer_2d_model.hpp"
+
+#include <fstream>
+
+#include "json_utils.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace genai {
+
+size_t get_vae_scale_factor(const std::filesystem::path& vae_config_path);
+
+FluxTransformer2DModel::Config::Config(const std::filesystem::path& config_path) {
+    std::ifstream file(config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", config_path);
+
+    nlohmann::json data = nlohmann::json::parse(file);
+    using utils::read_json_param;
+
+    read_json_param(data, "in_channels", in_channels);
+
+    file.close();
+
+    // block_out_channels should be read from VAE encoder / decoder config to compute proper m_vae_scale_factor
+    std::filesystem::path vae_config_path = config_path.parent_path().parent_path() / "vae_decoder" / "config.json";
+    file.open(vae_config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", vae_config_path);
+    data = nlohmann::json::parse(file);
+    read_json_param(data, "block_out_channels", block_out_channels);
```

Comment:
looks like `block_out_channels` is not used at all, because `m_vae_scale_factor` is computed via `get_vae_scale_factor`